crates/sui-core/src/authority/authority_test_utils.rs

```diff
@@ -16,81 +15,175 @@ use super::*;
 #[cfg(test)]
 use super::shared_object_version_manager::{AssignedTxAndVersions, Schedulable};
 
-pub async fn send_and_confirm_transaction(
+// =============================================================================
+// MFP (Mysticeti Fast Path) Test Helpers
+//
+// The MFP transaction flow is:
+//   1. Client signs transaction
+//   2. Validator validates transaction and acquires locks (voting phase)
+//   3. Transaction submitted to consensus
+//   4. Consensus orders and certifies the transaction
+//   5. Transaction comes out of consensus ready for execution
+//   6. Transaction is executed
+//
+// =============================================================================
+
+/// Validates a transaction and acquires locks on owned objects.
+/// This is the MFP "voting" phase - similar to what happens when a validator
+/// receives a transaction before submitting to consensus.
+///
+/// Returns the verified transaction ready for consensus submission.
+pub fn vote_transaction(
     authority: &AuthorityState,
     transaction: Transaction,
-) -> Result<(CertifiedTransaction, SignedTransactionEffects), SuiError> {
-    send_and_confirm_transaction_(
-        authority,
-        None, /* no fullnode_key_pair */
-        transaction,
-        false, /* no shared objects */
-    )
-    .await
+) -> Result<VerifiedTransaction, SuiError> {
+    let epoch_store = authority.load_epoch_store_one_call_per_task();
+    transaction.validity_check(&epoch_store.tx_validity_check_context())?;
+    let verified_tx = epoch_store
+        .verify_transaction_require_no_aliases(transaction)?
+        .into_tx();
+
+    // Validate the transaction and acquire locks on owned objects
+    authority.handle_vote_transaction(&epoch_store, verified_tx.clone())?;
+
+    Ok(verified_tx)
 }
-pub async fn send_and_confirm_transaction_(
+
+/// Creates a VerifiedExecutableTransaction from a signed transaction.
+/// This validates the transaction, votes on it, and creates an executable
+/// as if it came out of consensus.
+pub fn create_executable_transaction(
     authority: &AuthorityState,
-    fullnode: Option<&AuthorityState>,
     transaction: Transaction,
-    with_shared: bool, // transaction includes shared objects
-) -> Result<(CertifiedTransaction, SignedTransactionEffects), SuiError> {
-    let (txn, effects, _execution_error_opt) = send_and_confirm_transaction_with_execution_error(
-        authority,
-        fullnode,
-        transaction,
-        with_shared,
-        true,
-    )
-    .await?;
-    Ok((txn, effects))
+) -> Result<VerifiedExecutableTransaction, SuiError> {
+    let epoch_store = authority.load_epoch_store_one_call_per_task();
+    let verified_tx = vote_transaction(authority, transaction)?;
+    Ok(VerifiedExecutableTransaction::new_from_consensus(
+        verified_tx,
+        epoch_store.epoch(),
+    ))
 }
 
-pub async fn certify_transaction(
+/// Submits a transaction to consensus for ordering and version assignment.
```

Comment:
Maybe emphasize that this only simulates consensus submission, by assigning version.